src/ripple/rpc/json_body.h

```diff
@@ -57,19 +59,50 @@ struct json_body
         }
 
         void
-        init(beast::error_code&) noexcept
+        init(boost::beast::error_code&) noexcept
         {
         }
 
         boost::optional<std::pair<const_buffers_type, bool>>
-        get(beast::error_code& ec)
+        get(boost::beast::error_code& ec)
         {
             return {{buffer_.data(), false}};
         }
 
         void
-        finish(beast::error_code&)
+        finish(boost::beast::error_code&)
+        {
+        }
+    };
+
+    class writer
+    {
+        std::string body_string_;
+
+    public:
+        using const_buffers_type =
+            boost::asio::const_buffer;
+
+        template<bool isRequest, class Fields>
+        explicit
+        writer(boost::beast::http::message<isRequest,
+                json_body, Fields> const& msg)
+                : body_string_(to_string(msg.body()))
+        {
+        }
+
+        void
+        init(boost::beast::error_code& ec)
+        {
+            ec.assign(0, ec.category());
+        }
+
+        boost::optional<std::pair<const_buffers_type, bool>>
+        get(boost::beast::error_code& ec)
         {
+            ec.assign(0, ec.category());
+            return {{const_buffers_type{
+                body_string_.data(), body_string_.size()}, false}};
         }
```

Comment:
Looks good. I wonder if storing the buffer might be a better option.

eg.
```
    class writer
    {
        boost::asio::const_buffer buffer_;

    public:
        using const_buffers_type =
            boost::asio::const_buffer;

        template<bool isRequest, class Fields>
        explicit
        writer(boost::beast::http::message<isRequest,
                json_body, Fields> const& msg)
        {
            auto s {to_string(msg.body())};
            buffer_ = const_buffers_type{s.data(), s.size()};
        }

        void
        init(boost::beast::error_code& ec)
        {
            ec.assign(0, ec.category());
        }

        boost::optional<std::pair<const_buffers_type, bool>>
        get(boost::beast::error_code& ec)
        {
            ec.assign(0, ec.category());
            return {{buffer_, false}};
        }
    };
```

Reply:
`const_buffer` doesn't own it's memory, so I don't think that works (it would point to the re-claimed string memory).

Reply:
You're correct, I forgot const_buffer doesn't own the memory.